src/PanoImageRenderer/renderer/CylinderRenderer.js

```diff
@@ -29,27 +29,25 @@ export default class CylinderRenderer extends Renderer {
 
 	getVertexShaderSource() {
 		return `
-			attribute vec3 aVertexPosition;
-			attribute vec2 aTextureCoord;
-			uniform mat4 uMVMatrix;
-			uniform mat4 uPMatrix;
-			varying highp vec2 vTextureCoord;
-			void main(void) {
-				gl_Position = uPMatrix * uMVMatrix * vec4(aVertexPosition, 1.0);
-				vTextureCoord = aTextureCoord;
-			}`;
+attribute vec3 aVertexPosition;
+attribute vec2 aTextureCoord;
+uniform mat4 uMVMatrix;
+uniform mat4 uPMatrix;
+varying highp vec2 vTextureCoord;
+void main(void) {
+	vTextureCoord = aTextureCoord;
+	gl_Position = uPMatrix * uMVMatrix * vec4(aVertexPosition, 1.0);
+}`;
```

Comment:
@WoodNeck 

CubeRenderer, CubeStripRenderer, CylinderRenderer

The three vertaxShaders look the same.

Reply:
That's right, the legacy code actually has the same shader code.
I think that we have to extract them as a single file and use the webpack / rollup glsl shader loader instead, so we can reuse them easily.
What I want to say, again, is we have to do that when we start refactoring (maybe when we add the Typescript support..)